codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -421,9 +422,8 @@ public void processFragment(boolean endOfHeaders, ByteBuf fragment,
                     final HeadersBlockBuilder hdrBlockBuilder = headersBlockBuilder();
                     hdrBlockBuilder.addFragment(fragment, ctx.alloc(), endOfHeaders);
                     if (endOfHeaders) {
-                        listener.onHeadersRead(ctx, headersStreamId, hdrBlockBuilder.headers(),
-                                        streamDependency, weight, exclusive, padding, headersFlags.endOfStream());
-                        close();
+                        listener.onHeadersRead(ctx, headersStreamId, hdrBlockBuilder.headers(), streamDependency,
+                                weight, exclusive, padding, headersFlags.endOfStream());
```

Comment:
I'm assuming you removed the call to `close()` because it's automatically done by `headers()`?